File: src/test/java/hello/HelloCommandEmbed.java
package hello;

import io.reactivex.rxjava3.annotations.NonNull;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import dev.alphaserpentis.coffeecore.commands.BotCommand;
import dev.alphaserpentis.coffeecore.data.bot.CommandResponse;

public class HelloCommandEmbed extends BotCommand<MessageEmbed, SlashCommandInteractionEvent> {

    public HelloCommandEmbed() {
        super(
                new BotCommandOptions()
                        .setName("helloembed")
                        .setDescription("Says hello to you but in an embed!")
                        .setOnlyEmbed(true)
                        .setDeferReplies(true)
        );
    }

    @Override
    @NonNull
    public CommandResponse<MessageEmbed> runCommand(
            long userId,
            @NonNull SlashCommandInteractionEvent event
    ) {
        EmbedBuilder eb = new EmbedBuilder();

        eb.setTitle("Hello!");
        eb.setDescription("Hello, " + event.getUser().getAsMention() + "!");
        eb.setColor(0x00FF00);

        return new CommandResponse<>(isOnlyEphemeral(), eb.build());
    }
}

File: src/test/dev/alphaserpentis/examples/coffeecore/kotlin/hello/HelloCommandEmbed.kt
package hello

import dev.alphaserpentis.coffeecore.commands.BotCommand
import dev.alphaserpentis.coffeecore.data.bot.CommandResponse
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.entities.MessageEmbed
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class HelloCommandEmbed: BotCommand<MessageEmbed, SlashCommandInteractionEvent>(
    BotCommandOptions().apply {
        setName("helloembed")
        setDescription("Says hello to you but in an embed!")
        setOnlyEmbed(true)
        setDeferReplies(true)
    }
) {
    override fun runCommand(userId: Long, event: SlashCommandInteractionEvent): CommandResponse<MessageEmbed> {
        val eb = EmbedBuilder().setTitle("Hello!").setDescription("Hello, ${event.user.asMention}!").setColor(0x00FF00)

        return CommandResponse(isOnlyEphemeral, eb.build())
    }
}